services/src/main/java/org/jboss/windup/web/services/service/AnalysisContextService.java

```diff
@@ -39,16 +43,35 @@ public AnalysisContext createDefaultAnalysisContext()
     public AnalysisContext create(AnalysisContext analysisContext)
     {
         this.ensureSystemRulesPathsPresent(analysisContext);
+        this.loadPackagesToAnalysisContext(analysisContext);
         entityManager.persist(analysisContext);
+
         return analysisContext;
     }
 
+    protected void loadPackagesToAnalysisContext(AnalysisContext analysisContext)
+    {
+        analysisContext.setIncludePackages(this.loadPackagesFromPersistenceContext(analysisContext.getIncludePackages()));
+        analysisContext.setExcludePackages(this.loadPackagesFromPersistenceContext(analysisContext.getExcludePackages()));
+    }
+
+    protected Set<Package> loadPackagesFromPersistenceContext(Collection<Package> detachedPackages)
+    {
+        return detachedPackages.stream()
+                .filter(aPackage -> aPackage.getId() != null && aPackage.getId() != 0)
+                .map(aPackage -> this.entityManager.find(Package.class, aPackage.getId()))
+                .filter(aPackage -> aPackage != null)
+                .collect(Collectors.toSet());
+    }
```

Comment:
@jsight What do you think about this part? I need to ensure package entities come from persistence context. But I'm not sure what should I do in case user sends invalid package? (Package without id or with non existing id). Right now I juts ignore it. Should I keep it that way, or throw some exception?